README.md

```diff
@@ -1,491 +1,50 @@
-    </td>
-    <td><tt>nil</tt></td>
-  </tr>
-  <tr>
-    <td><tt>['consul']['serve_ui']</tt></td>
-    <td>Boolean</td>
-    <td>Determines whether the consul service also serve's the UI</td>
-    <td><tt>false</tt></td>
-  </tr>
-</table>
-
-## Usage
-The easiest way to bootstrap a cluster is to use the cluster recipe
-and use [Chef provisioning][8] which is a relatively new
-extension. This extension allows you to use any driver and easily
-stand up a cluster. Once the [Chef Development Kit][9] has been
-installed you can run the following command to provision a cluster.
-
+This cookbook provides node attributes which can be used to fine tune
+how the recipes install and configure the Consul client, server and
+UI. These values are passed into the resource/providers for
+validation prior to converging.
+
+|   Key   |  Type  |   Description  |  Default  |
+|---------|--------|----------------|-----------|
+| ['consul']['version'] | String | Installation version | 0.5.1 |
+| ['consul']['remote_url'] | String | Remote URL for download. | https://dl.bintray.com/mitchellh/consul |
+| ['consul']['service_name'] | String | Name of the service (operating system) | consul |
+| ['consul']['service_user'] | String | Name of the service user | consul |
+| ['consul']['service_group'] | String | Name of the service group | consul |
+
+## Resources/Providers
+This cookbook provides resource and provider primitives to manage the
+Consul client, server and UI. These primitives are what is used in the
+recipes, and should be used in your own [wrapper cookbooks][2].
+
+### consul_config
 ```ruby
-gem install chef-provisioning chef-provisioning-fog
-export CHEF_DRIVER=fog:AWS
-chef-client -z cluster.rb
+consul_config '/etc/consul.json' do
+  user 'consul'
+  group 'consul'
+end
 ```
-
-Please follow the [Chef provisioning README][10] which provides more
-detailed information about provisioning. You'll need to configure
-your credentials prior to provisioning.
-
-### consul::default
-The default recipe will install the Consul agent using the
-`consul::install_binary` recipe. It will also configure and
-start consul at the machine boot.
-
-### consul::install_binary
-If you only wish to simply install the binary from the official
-mirror you simply include `consul::install_binary` in your node's
-`run_list`:
-
-```json
-{
-  "run_list": [
-    "recipe[consul::install_binary]"
-  ]
-}
-```
-
-### consul::install_source
-Instead if you wish to install Consul from source you simply need
-to include `consul::install_source` in your node's `run_list`. This
-will also configure the Go language framework on the node to build
-the application.
-
-```json
-{
-  "run_list": [
-    "recipe[consul::install_source]"
-  ]
-}
-```
-
-### consul::ui
-Installing the separate Consul UI simply requires you to include
-the `consul::ui` recipe in your node's `run_list`.
-
-```json
-{
-  "run_list": [
-    "recipe[consul::ui]"
-  ]
-}
+### consul_definition
+### consul_service
```

Comment:
I think this should be the first documented LWRP and immediately follow a quick "usage" section. Usage should include instructions for including the default recipe and explain what you get and also how to leverage the consul-service lwrp